packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/TypeProvider.cs

```diff
@@ -71,35 +93,160 @@ private TypeSignatureModifiers GetDeclarationModifiersInternal()
             return modifiers;
         }
 
-        public CSharpType? Inherits { get; protected init; }
+        public CSharpType? Inherits
+        {
+            get => _innerProvider == this ? _inherits : _innerProvider.Inherits;
+            protected set
```

Comment:
For these protected init setters, I think we can remove the else path as the _innerProvider should always be `this`. Then we can add back `init`.